proposals/XX-bw-xxx.txt

```diff
@@ -0,0 +1,178 @@
+          Tor Bandwidth Measurements Document Format
+                            juga
+
+1. Scope and preliminaries
+
+  This document describes the format of Tor's bandwidth measurements
+  document, version 1.1.0 and later.
+
+  Since Tor version 0.2.4.12-alpha the directory
+  authorities use the bandwidth measurements document called
+  "V3BandwidthsFile" and produced by Torflow [1]
+  (format described in README.spec.txt [2]).
+
+    The key words "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL
+    NOT", "SHOULD", "SHOULD NOT", "RECOMMENDED",  "MAY", and
+    "OPTIONAL" in this document are to be interpreted as described in
+    RFC 2119.
+
+1.2. Acknowledgements
+
+  The original bandwidth measurement scanner (Torflow) and format was
+  created by mike. Teor suggested to write this specification while
+  contributing on pastly's new bandwidth scanner implementation.
+
+  This specification was revised after feedback from:
+
+    Teor
+
+1.3 Outline
+
+  The bandwidth measurements mentioned in sections 3.4.1 and 3.4.2
+  of "Tor directory protocol" (dir-spec.txt) [3] are obtained
+  by bandwidth authorities, which generate a file storing information
+  on relays' measured bandwidth capacities.
+
+2. Format details
+
+  Bandwidth measurements MUST contain the following sections:
+  - Header (exactly once)
+  - Relays measurements (zero or more times)
+
+2.1. Definitions
+
+  The following nonterminals are defined in dir-spec.txt, sections
+  1.2., 2.1.1., 2.1.3. :
+
+    Int
+    SP (space)
+    NL (newline)
+    Keyword
+    ArgumentChar
+    fingerprint (hexdigest)
+    nickname
+
+  Nonterminals defined in "Tor Directory List Format" (dir-list-spec.txt),
+  section 2.2.1.:
+
+    version_number
+
+  We define the following nonterminals:
+
+    value ::= ArgumentChar+
+    key_value ::= Keyword "=" value
+    line := ArgumentChar* NL
+    timestamp = Int
+    bandwidth = Int
+
+2.2. Header format
+
+It consists of:
+
+  timestamp NL
+
+    [At start, exactly once.]
+
+    The Unix Epoch time in seconds when the file was created.
+
+  "version="version_number NL
+
+    [In second position, zero or one time.]
+
+    The specification document format version.
+
+    New implementations SHOULD include this line.
+    For documents generated following this specification,
+    the version_number MUST be "1.1.0".
+    Documents generated before this specification do not contain this line,
+    and the version_number is considered to be "1.0.0".
+    This line is ignored by any Tor version released before this specification.
+    TODO: implement this feature in tor
+
+  "software="Keyword NL
```

Comment:
Use value, not Keyword.

It is ok to write definitions with a space, if you want. Spaces outside "" and SP do not appear in the output:
```
"software=" value NL
```